rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -271,8 +271,8 @@ class Subscription : public SubscriptionBase
 
     if (subscription_topic_statistics_) {
       const auto nanos = std::chrono::time_point_cast<std::chrono::nanoseconds>(
-        std::chrono::steady_clock::now());
-      const auto time = rclcpp::Time(nanos.time_since_epoch().count(), RCL_STEADY_TIME);
+        std::chrono::system_clock::now());
```

Comment:
IIUC, this change is because we're assuming that the timestamp in message headers is using system time, right?

What happens if a node is using simulation time (ie. ROS_TIME)?

Reply:
That is correct. There is also no way to specify a message's time source when we were assuming one would use the std_msgs/Header (I brought this up with @wjwwood and @tfoote during the message review).

Reply:
Okay, so it sounds like a known limitation of topic statistics for the moment; not compatible with simulation.

Reply:
The type in `std_msgs::msg::Header` can be ROS time... Both ROS Time and System time share an epoch, the unix epoch. It's only when you want to use a different Epoch or behavior for the clock that the stamp in header is insufficient, e.g. like steady time or TAI time.

So in this situation I don't  understand why you cannot use ROS Time...

Reply:
I guess we could ROS time, and assume the incoming messages are using the same epoch as this node.

At the end of the day we can't know if the time in the header will have the same epoch as the time from this node, right? Since one of them may being use simulation time and the other not.

Reply:
Simulated time uses the same epoch as system time, at least that's my understanding. If you are simulating time and then set the time to 1 (as in 1 nanosecond) then it's 1 nanosecond after jan 1 1970.

At least that's how I interrpret it:

- this is the rcl time point:
  - https://github.com/ros2/rcl/blob/643f6ec5300a24b8f9076019232f98b8d6aa76b6/rcl/include/rcl/time.h#L43-L44
- which the rclcpp::Time is based on:
  - https://github.com/ros2/rclcpp/blob/df3c2ffa8a9d0b1b6f5cf707986accbc091575f9/rclcpp/include/rclcpp/time.hpp#L129

Perhaps it should be clarified, but I think ROS time and system time should be usable interchangeably.